bedrock/base/monkeypatches.py

```diff
@@ -0,0 +1,32 @@
+import logging
+
+
+__all__ = ['patch']
+
+# Idempotence! http://en.wikipedia.org/wiki/Idempotence
+_has_patched = False
+
+
+def patch():
+    global _has_patched
```

Comment:
I know you're just copying this over from funfactory, but that doesn't make me dislike this whole function and any less :/
Removing the need for these monkeypatches is outside of the scope of this PR, but I hope we get a chance to eliminate it eventually.